### DISCUSSION TITLE
feat: add `eslint-plugin-sonarjs`

### DISCUSSION THREAD

**Discussion by zanminkian** [antfu/eslint-config#497] [Ideas]
This plugin looks great for a better code quality. Thought?

**Comment by phoenix-oes** [comment]
I [just](https://github.com/antfu/eslint-config/discussions/675) went through this myself. Here's how you can add it yourself for your own projects:

At the top, use [FlatCompat](https://eslint.org/docs/latest/use/configure/migration-guide#using-eslintrc-configs-in-flat-config)

```js
import { FlatCompat } from '@eslint/eslintrc';

const compat = new FlatCompat();
```

At the bottom, beneath all other configs, add one new legacy config:

```js
  // Legacy config
  ...compat.config({
    extends: [
      'plugin:sonarjs/recommended-legacy',
    ],
    rules: {
      'sonarjs/todo-tag': 'warn',
      'sonarjs/no-commented-code': 'warn',
    },
  }),
);
```

Maybe not ideal, but I also like being able to quickly toggle it off for certain situations.

**Reply by EugenGedroyc** [reply]
The flat configuration is ready now
```
{
  name: 'sonar',
  ...pluginSonar.configs.recommended
},
```
